modules/cancerBiomarkers.py

```diff
@@ -0,0 +1,352 @@
+        )
+
+        # Write evidence strings
+        self.write_evidence_strings(output_file)
+        logging.info(f'{evidence.count()} evidence strings have been saved to {output_file}.')
+
+    def compute_biomarkers(
+        self,
+        biomarkers_df: DataFrame,
+        source_df: DataFrame,
+        disease_df: DataFrame,
+        drugs_df: DataFrame
+    ) -> DataFrame:
+        """The diverse steps to prepare and enrich the input table are computed"""
+
+        biomarkers_enriched = (
+            biomarkers_df
+            .select(
+                'Biomarker', 'IndividualMutation',
+                array_distinct(split(col('Alteration'), ';')).alias('alterations'),
+                array_distinct(split(col('Gene'), ';')).alias('gene'),
+                split(col('AlterationType'), ';').alias('alteration_types'),
+                array_distinct(split(col("PrimaryTumorTypeFullName"), ";")).alias('tumor_type_full_name'),
+                array_distinct(split(col('Drug'), ';')).alias('drug'),
+                'DrugFullName', 'Association', 'EvidenceLevel', 'gDNA',
+                array_distinct(split(col('Source'), ';')).alias('source')
+            )
+            .withColumn('tumor_type_full_name', explode(col('tumor_type_full_name')))
+            .withColumn('tumor_type', translate(col('tumor_type_full_name'), ' -', ''))
+            .withColumn('gene', explode(col('gene')))
+            .withColumn('drug', explode(col('drug')))
+            # Override specific genes
+            .withColumn(
+                'gene',
+                when(col('gene') == 'C15orf55', 'NUTM1')
+                .when(col('gene') == 'MLL', 'KMT2A')
+                .when(col('gene') == 'MLL2', 'KMT2D')
+                .otherwise(col('gene'))
+            )
+            .withColumn('gene', upper(col('gene')))
+            # Disambiguate alteration_type when biomarker consists of multiple alterations
+            .withColumn(
+                'tmp',
+                self.zip_alterations_with_type_udf(col('alterations'), col('alteration_types')))
+            .withColumn('tmp', explode(col('tmp')))
+            .withColumn('alteration_type', element_at(col('tmp'), 2))
+            .withColumn(
+                'alteration',
+                when(
+                    ~col('IndividualMutation').isNull(),
+                    col('IndividualMutation')
+                )
+                .otherwise(element_at(col('tmp'), 1))
+            )
+            .drop('tmp')
+            # Clean special cases on the alteration string
+            .withColumn(
+                'alteration',
+                when(
+                    col('alteration').contains(':.'),
+                    translate(col('alteration'), ':.', '')
+                )
+                .when(
+                    col('alteration') == 'NRAS:.12.,.13.,.59.,.61.,.117.,.146.',
+                    col('Biomarker')  # 'NRAS (12,13,59,61,117,146)'
+                )
```

Comment:
I think if `col('alteration').contains(':.')` in row 143, this condition will never be satisfied.